File: ruby/my_service.rb
# Imports
require 'aws_iot_device'

# Configuration
host = "a267zn9knxsui0-ats.iot.eu-west-1.amazonaws.com"
port = 8883
certificate_path = "../certs/certificate.pem.crt"
private_key_path = "../certs/private.pem.key"
root_ca_path = "../certs/root-CA.pem"
topic = "uservicehack/kittens"
service_name = 'ruby_kittens'

# MQTT client
client = AwsIotDevice::MqttShadowClient::MqttManager.new(host: host, port: port, ssl: true)
client.config_ssl_context(root_ca_path, private_key_path, certificate_path)
client.connect

# Helpers
callback = Proc.new do |message|
  puts "On topic: #{message.topic}, message: #{message.payload}"
end

# Main
#
# Register a callback message logger on our topic 
client.subscribe("uservicehack/+", 0, callback)

# Loop forever publishing a new message to topic every three seconds
loop do
  client.publish(topic, "time in rubyland is now #{Time.now.to_i}")
  sleep 3
end

# Use 'client.disconnect' if you don't want to loop forever

File: node/my_service.js
// An example MQTT service that publishes and subscribes
// Uses the AWS IoT messaging service
// Libraries
var aws_iot = require('aws-iot-device-sdk')

// Constants
const EVERY_THREE_SECONDS = 3000

// Configuration
const service_name = 'node_kittens'
const aws_iot_config = {
  keyPath: "../certs/private.pem.key",
  certPath: "../certs/certificate.pem.crt",
  caPath: "../certs/root-CA.pem",
  clientId: service_name,
  host: "a267zn9knxsui0-ats.iot.eu-west-1.amazonaws.com",
  debug: false
}
const client = aws_iot.device(aws_iot_config)

// Helpers
const get_time = () => parseInt((new Date().getTime() / 1000).toFixed(0))

// Main
client.on('connect', () => {
  // The '#' wildcard matches all topics below this one
  client.subscribe('workshop/#', function (err) {
    if (!err) {
      client.publish('workshop/kittens', `Service: ${service_name} has connected!`)
      console.log('Press Control-C to quit.')
    }
  })
  setInterval(() => {
    client.publish('workshop/kittens',
                   `At the third stroke, the time will be: ${get_time()}`)
  }, EVERY_THREE_SECONDS)
})

client.on('message', (topic, message) => {
  // message is of type Buffer, so we need to use .toString
  console.log(`Message on topic: ${topic} is: ${message.toString()}`)
  // client.end() // comment this to loop forever
})
